### DISCUSSION TITLE
Reference a file relative

### DISCUSSION THREAD

**Discussion by ptr1120** [hairyhenderson/gomplate#1363] [Q&A] [answered]
Hello,

 is the a way to reference/read a file relative to the current template file? I need this in order to load a file with variables that are specific to the templates of this subdirectory. 
So a function/variable like `{{file.Read currentFile.Directory/.env }}`?

**Comment by hairyhenderson** [comment, marked as answer]
@ptr1120 is there a reason you need to use `file.Read`, rather than including the file as a datasource? `.env` type files are [natively supported](https://docs.gomplate.ca/datasources/#the-env-file-format) by gomplate...

If you _do_ need to use `file.Read` though, the path is relative to your current working directory (i.e. whatever the `pwd` command returns). I've thought in the past of some sort of `self` function, which could return the template's path, or something similar, but I'm not certain how necessary it really is... Every use-case I've run into for it turned out to be a symptom of me trying to do something in an awkward and inelegant way 😉

**Comment by ptr1120** [comment]
@hairyhenderson : first of all thank for this great project and for your answer. My use case ist that I have a lot of yml configuration files for several tenants. Imagine the following simplified file/directory structure

```
common/
   common-vars.env
tenants/
   tenant1/
     .env
     user-settings.yml
     apps/
        app1.yml
  tenant2/
     .env
     user-settings.yml
```

The .env files contain critical variables like the tenantId that is used in the other yml files. My problem is that I want to ensure, when referencing a tenantId in a yml, that it is the id from the .env file in this subdirectory tree. Using an absolute path would introduce the risk of failures due to copy and paste.
I am not aware of all the features/patterns in gomplate, but for me getting the name of the current file and then reading the env file relative from this would be something I would typically do. Such a `self.Path`, `thisFilePath` or `thisFileDirectory` function/variable would even give me the possibillity to generate the tenantId from the given path which would ensure correctness/uniqueness in this special case.

**Reply by hairyhenderson** [reply]
(FYI I edited your text to put a code-fence around the directory structure so it's easier to see! 😅)

Ok, it sounds like this would be a useful feature then! I'll see what I can do. I'm not sure how quickly I'll be able to get to this, but it's on the list!

**Comment by ptr1120** [comment]
thank you and sorry I wrote it from my mobile phone.

**Comment by hairyhenderson** [comment]
I realized that this probably belongs in the [`tmpl` namespace](https://docs.gomplate.ca/functions/tmpl/), so I've issued https://github.com/hairyhenderson/gomplate/pull/1364 to add two new functions `tmpl.Path` and `tmpl.PathDir` - the second is basically the same as `filepath.Dir tmpl.Path`, though the behaviour is slightly different when the template is provided inline (via `-i`).

**Comment by ptr1120** [comment]
Looks very promising, nice that you also provide both functions `tmpl.Path` and `tmpl.PathDir`
